Intersphinx support (#86)
readthedocs/sphinx-hoverxref: docs/configuration.rst

          'setting',
      ]

.. confval:: hoverxref_intersphinx

   Description: Whether or not enable Sphinx's hoverxref extension on intersphinx links.

   Default: ``False``

   Type: bool

> .. confval:: hoverxref_intersphinx_type

Author: Good point, `style` sounds better! Unfortunately, I've already made the choice for `_type` in the other configs 😞 

- hoverxref_role_types
- hoverxref_default_type

I think it makes sense to continue with the same pattern, or change them altogether.

.. confval:: hoverxref_intersphinx

   Description: Enable Sphinx's hoverxref extension on intersphinx targets from ``intersphinx_mapping``.

   Default: ``[]``

   Type: list

   .. warning::

      The Sphinx's target project **must be hosted on Read the Docs** to work.

>       This is a current limitation that we may be able to remove in the future.

Reviewer: ```suggestion
      This is a current limitation that we hope to remove in the future.
```

> .. confval:: hoverxref_intersphinx_types

Reviewer: Is this useful? It seems you'd want to change the modal on content type (eg. `doc` or `class`) more than at the project-level?

Author: Good point!

In that case we would need a nested dictionary:

```python
hoverxref_intersphinx_types = {
 'python': {
    'ref': 'tooltip',
    'doc': 'modal',
  },
}
```

It's basically one [`hoverxref_role_types`](https://sphinx-hoverxref.readthedocs.io/en/latest/configuration.html#confval-hoverxref_role_types) per intersphinx mapping.

However, I'm not sure how easy is to detect the role used in the missing-reference. I'll take a look and come back 😄

Author: I found how to this. I pushed the changes already.
